server/applicationPattern/applicationPattern/onfModel/services/ForwardingConstructAutomationServices.test.js

```diff
@@ -0,0 +1,412 @@
+const ForwardingConstructAutomationServices = require('./ForwardingConstructAutomationServices');
+const ControlConstruct = require('../models/ControlConstruct');
+const ForwardingDomain = require('../models/ForwardingDomain');
+const event = require('../../rest/client/eventDispatcher');
+const fileOperation = require('../../databaseDriver/JSONDriver');
+
+const ltps = [
+  {
+    "uuid": "ro-2-0-1-op-s-bm-001",
+    "ltp-direction": "core-model-1-4:TERMINATION_DIRECTION_SOURCE",
+    "client-ltp": [
+    ],
+    "server-ltp": [
+      "ro-2-0-1-http-s-000"
+    ],
+    "layer-protocol": [
+      {
+        "local-id": "0",
+        "layer-protocol-name": "operation-server-interface-1-0:LAYER_PROTOCOL_NAME_TYPE_OPERATION_LAYER",
+        "operation-server-interface-1-0:operation-server-interface-pac": {
+          "operation-server-interface-capability": {
+            "operation-name": "/v1/embed-yourself"
+          },
+          "operation-server-interface-configuration": {
+            "life-cycle-state": "operation-server-interface-1-0:LIFE_CYCLE_STATE_TYPE_EXPERIMENTAL",
+            "operation-key": "Operation key not yet provided."
+          }
+        }
+      }
+    ]
+  }
+];
+
+const ltp = ltps[0];
+
+const fcArray = [
+  {
+    "uuid": "ro-2-0-1-op-fc-bm-001",
+    "name": [
+      {
+        "value-name": "ForwardingKind",
+        "value": "core-model-1-4:FORWARDING_KIND_TYPE_INVARIANT_PROCESS_SNIPPET"
+      },
+      {
+        "value-name": "ForwardingName",
+        "value": "PromptForEmbeddingCausesRequestForBequeathingData"
+      }
+    ],
+    "fc-port": [
+      {
+        "local-id": "100",
+        "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_INPUT",
+        "logical-termination-point": "ro-2-0-1-op-s-bm-001"
+      },
+      {
+        "local-id": "200",
+        "port-direction": "core-model-1-4:PORT_DIRECTION_TYPE_OUTPUT",
+        "logical-termination-point": "ro-2-0-1-op-c-bm-or-1-0-0-000"
+      }
+    ]
+  }
+];
+
+const fc = fcArray[0];
+
+const forwardingAutomationInputList = [
+  {
+    forwardingName: 'PromptForEmbeddingCausesRequestForBequeathingData',
+    attributeList: {
+      'new-application-name': 'RegistryOffice',
+      'new-application-release': '2.0.1',
+      'new-application-protocol': 'HTTP',
+      'new-application-address': { 'ipv-4-address': '1.1.3.8' },
+      'new-application-port': 3008
+    },
+    context: undefined
+  },
+  {
+    forwardingName: 'ServiceRequestCausesLtpUpdateRequest',
+    attributeList: {
```

Comment:
AttributeList has 0 information, the parameter is passed from automateForwardingConstruct to eventDispatcher without any alternation, it can safely be an empty array.